src/OrleansSQLUtils/CreateOrleansTables_MySql.sql

```diff
@@ -441,77 +408,26 @@ BEGIN
 			)
 			VALUES
 			(
-				_serviceId,
-				_grainId,
-				_reminderName,
-				_startTime,
-				_period,
-				_grainIdConsistentHash
-			);
-		ELSE
-			UPDATE OrleansRemindersTable
-			SET				
-				StartTime             = _startTime,
-				Period                = _period,
-				GrainIdConsistentHash = _grainIdConsistentHash,
-				ETag                  = _newEtag
-			WHERE
-				ServiceId = _serviceId AND _serviceId IS NOT NULL
-				AND GrainId = _grainId AND _grainId IS NOT NULL
-				AND ReminderName = _reminderName AND _reminderName IS NOT NULL;
-		END IF;
-		COMMIT;
-		SELECT CONVERT(_newEtag, BINARY) AS ETag;
-END$$
-
-DELIMITER ;
+				@serviceId,
+				@grainId,
+				@reminderName,
+				@startTime,
+				@period,
+				@grainIdConsistentHash
+			)
+		ON DUPLICATE KEY UPDATE 
+				StartTime             = @startTime,
+				Period                = @period,
+				GrainIdConsistentHash = @grainIdConsistentHash,
+				ETag                  = last_insert_id(ETag+1);
+	
+		SELECT CONVERT(last_insert_id(), BINARY) AS ETag;
```

Comment:
This looks like working like `ROWVERSION` in SQL Server (cf. http://www.informit.com/articles/article.aspx?p=376253&seqNum=4), clever.